rust/kcl-lib/src/parsing/ast/types/mod.rs

```diff
@@ -2396,6 +2396,28 @@ impl Name {
     }
 }
 
+impl Name {
+    /// Write the full name to the given string.
+    pub fn write_to(&self, buf: &mut String) {
+        if self.abs_path {
+            buf.push_str("::");
+        };
+        for p in &self.path {
+            buf.push_str(&p.name);
+            buf.push_str("::");
+        }
+        buf.push_str(&self.name.name);
+    }
+}
+
+impl fmt::Display for Name {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut buf = String::new();
+        self.write_to(&mut buf);
```

Comment:
This seems like an unneeded string allocation when using `write!()`. Is there a way we could get the best of both worlds?

Reply:
We use this when we call `identifer.to_string()`.

Reply:
`write_to` should take an `impl fmt::Write` rather than a concrete `String` then we can just pass `f` here